src/ecalc/libraries/libecalc/common/libecalc/expression/expression_evaluator.py

```diff
@@ -100,11 +99,21 @@ def eval_parenteses(
                 ) from e
 
             tokens = tokens[:substart] + [tokens_evaluated] + tokens[subend + 1 :]
+            number_of_left_parentheses, number_of_right_parentheses = count_parentheses(tokens=tokens)
 
     return eval_logicals(tokens)
 
 
+def count_parentheses(tokens: List[Union[float, int, bool, NDArray[np.float64], str]]) -> Tuple[int, int]:
```

Comment:
nice to extract it into a function. Makes it much easier to test it in isolation as well. *hint* :)